source/dbo/Stored Procedures/Power BI Reporting/uspNOIDSWeeklyCount.sql

```diff
@@ -0,0 +1,23 @@
+﻿/*on a Monday morning, count the number of rows in the NOIDS view
+for the week before and store this in a table */
+CREATE PROCEDURE [dbo].[uspGenerateNOIDSWeeklyCount]
+AS
+	DECLARE @ThisWeek VARCHAR(10) =  (SELECT ISOYearWeek FROM [dbo].Calendar WHERE DateValue = CONVERT(DATE, GETUTCDATE()))
+	DECLARE @LastSunday DATE =  (SELECT MAX(DateValue) FROM [dbo].Calendar WHERE DateValue <= GETUTCDATE() AND DATENAME(dw, DateValue) = 'Sunday')
+
+	--check to see if a value has already been stored for this week
+	IF NOT EXISTS
+	(SELECT wc.DateValue 
+		FROM [dbo].[NOIDSWeeklyCount] wc
+			INNER JOIN [dbo].[Calendar] c ON c.DateValue = wc.DateValue
```

Comment:
This join is confusing me... it looks like `DateValue` in `Calendar` is just a date, whereas `DateValue` in `NOIDSWeeklyCount` is the exact time of a previous run, so this won't ever match anything?